x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/StEnvelope.java

```diff
@@ -110,37 +131,82 @@ public DataType dataType() {
 
     @Override
     public Expression replaceChildren(List<Expression> newChildren) {
-        return new StEnvelope(source(), newChildren.get(0));
+        return new StEnvelope(source(), newChildren.getFirst(), spatialDocValues);
     }
 
     @Override
     protected NodeInfo<? extends Expression> info() {
-        return NodeInfo.create(this, StEnvelope::new, field());
+        return NodeInfo.create(this, StEnvelope::new, spatialField());
     }
 
-    @ConvertEvaluator(extraName = "FromWKB", warnExceptions = { IllegalArgumentException.class })
```

Comment:
We've changed the `ConvertEvaluator` to `Evaluator`, which means before when we had a multivalue composed of several points, we would have outputted the same points?

Now we'll consider a point multivalue as a geometry collection, so we'll compute the envelope on those?